runtime/src/commonMain/kotlin/pbandk/internal/json/JsonValueDecoder.kt

```diff
@@ -118,15 +154,46 @@ internal class JsonValueDecoder(private val jsonConfig: JsonConfig) {
     }
 
     fun readFloat(value: JsonElement): Float = try {
-        // TODO handle Inf and NaN
-        value.float
+        val floatValue = value.float
+        when {
+            floatValue.isFinite() -> {
+                if (floatValue < FLOAT_MIN || floatValue > FLOAT_MAX) {
+                    throw NumberFormatException("value out of bounds")
+                }
+
+                floatValue
+            }
+            (value as JsonLiteral).isString -> {
+                floatValue
+            }
+            else -> {
+                throw IllegalArgumentException("non finite values must be quoted")
+            }
+        }
     } catch (e: Exception) {
         throw InvalidProtocolBufferException("float field did not contain a float value in JSON", e)
     }
 
     fun readDouble(value: JsonElement): Double = try {
-        // TODO handle Inf and NaN
-        value.double
+        val doubleValue = value.double
+        when {
+            doubleValue.isFinite() -> {
+                if (doubleValue > 0.0 && (doubleValue < DOUBLE_MIN_POSITIVE|| doubleValue > DOUBLE_MAX_POSITIVE)) {
+                    throw NumberFormatException("value out of bounds")
+                }
+                if (doubleValue < 0.0 && (doubleValue < DOUBLE_MIN_NEGATIVE || doubleValue > DOUBLE_MAX_NEGATIVE)) {
```

Comment:
Shouldn't the signs in this test be flipped? I.e. `doubleValue > DOUBLE_MIN_NEGATIVE || doubleValue < DOUBLE_MAX_NEGATIVE`

Reply:
For a range `-0.00001`..`-2000000.0`, `-0.2` is out of bounds since `-0.2 < MIN` (note the case only handles negative numbers)